src/FFmpegReader.cpp

```diff
@@ -2076,14 +2098,12 @@ bool FFmpegReader::CheckMissingFrame(int64_t requested_frame) {
 
 	// Special MP3 Handling (ignore more than 1 video frame)
 	if (info.has_audio and info.has_video) {
-		AVCodecID aCodecId = AV_FIND_DECODER_CODEC_ID(aStream);
-		AVCodecID vCodecId = AV_FIND_DECODER_CODEC_ID(pStream);
 		// If MP3 with single video frame, handle this special case by copying the previously
 		// decoded image to the new frame. Otherwise, it will spend a huge amount of
 		// CPU time looking for missing images for all the audio-only frames.
-		if (checked_frames[requested_frame] > 8 && !missing_video_frames.count(requested_frame) &&
+		if (!missing_video_frames.count(requested_frame) &&
 			!processing_audio_frames.count(requested_frame) && processed_audio_frames.count(requested_frame) &&
-			last_frame && last_video_frame->has_image_data && aCodecId == AV_CODEC_ID_MP3 && (vCodecId == AV_CODEC_ID_MJPEGB || vCodecId == AV_CODEC_ID_MJPEG)) {
+			last_video_frame && last_video_frame->has_image_data && HasAlbumArt()) {
```

Comment:
Same issue here as well, by removing the `AV_CODEC_ID_MP3` limiter.

Reply:
That's intentional. I want to make the logic work for other file types or image types.

Reply:
@PhysSong Understood, and it definitely needs to be removed if album art is going to be supported across audio files generally. (_**Most**_ of which, these days, are not MP3s with JPEG-encoded image payloads.) 

As long as `HasAlbumArt()` works for both audio and video files (returning the answer we want, in all the various input file cases), then I'm on board with removing as many of the other conditions as possible.